experimental/traffic-portal/src/app/core/cdns/cdn-table/cdn-table.component.ts

```diff
@@ -0,0 +1,248 @@
+		},
+		{
+			action: "delete",
+			multiRow: false,
+			name: "Delete",
+		},
+		{
+			action: "snapshot-diff",
+			multiRow: false,
+			name: "Diff Snapshot",
+		},
+		{
+			action: "queue",
+			multiRow: false,
+			name: "Queue Server Updates"
+		},
+		{
+			action: "dequeue",
+			multiRow: false,
+			name: "Clear Queued Updates"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/dnssec-keys` ,
+			name: "Manage DNSSEC Keys"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/federations` ,
+			name: "Manage Federations"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/delivery-services` ,
+			name: "Manage Delivery Services"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/profiles?cdnName=${selectedRow.id}` ,
+			name: "Manage Profiles"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/servers` ,
+			name: "Manage Servers"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/notifications` ,
+			name: "Manage Notifications"
+		},
+	];
+
+	/**
+	 * A subject that child components can subscribe to for access to the fuzzy
+	 * search query text.
+	 */
+	public fuzzySubject: BehaviorSubject<string>;
+
+	/** Form controller for the user search input. */
+	public fuzzControl: FormControl = new FormControl("");
+
+	constructor(
+		private readonly alerts: AlertService,
+		private readonly api: CDNService,
+		public readonly auth: CurrentUserService,
+		private readonly dialog: MatDialog,
+		private readonly route: ActivatedRoute,
+	) {
+		this.fuzzySubject = new BehaviorSubject<string>("");
+		this.cdns = this.api.getCDNs();
+	}
+
+	/** Initializes table data, loading it from Traffic Ops. */
+	public ngOnInit(): void {
+		this.route.queryParamMap.subscribe(
+			m => {
+				const search = m.get("search");
+				if (search) {
+					this.fuzzControl.setValue(decodeURIComponent(search));
+					this.updateURL();
+				}
+			}
+		);
+	}
+
+	/** Update the URL's 'search' query parameter for the user's search input. */
+	public updateURL(): void {
+		this.fuzzySubject.next(this.fuzzControl.value);
+	}
+
+	/**
+	 * Queues or clears updates on a group of CDNs.
+	 *
+	 * @param cdn The CDN on which to queue updates.
+	 * @param queue Whether updates should be queued (`true`) or cleared
+	 * (`false`).
+	 */
+	private async queueUpdates(cdn: ResponseCDN, queue: boolean = true): Promise<void> {
+		const title = `${queue ? "Queue" : "Clear"} Updates on ${cdn.name}?`;
+		const action = queue ? "queue" : "dequeue";
+		const ref = this.dialog.open<DecisionDialogComponent, DecisionDialogData, boolean>(DecisionDialogComponent, {
+			data: {
+				message: `Are you sure you want to ${action} server updates for all the ${cdn.name} servers?`,
```

Comment:
grammar: "... all <ins>of</ins> the ..."

Reply:
*of* added in cc7d8f3e66